docs/vm-compose.md

```diff
@@ -0,0 +1,157 @@
+# VM Configuration creation
+
+## Motivation
+
+TBD define use cases
+
+KubeVirt should allows users to start small, but also provide fully fledged
+VM definitions.
+In that sense a user should be able to just name a disk and for example the
+operating system which is provided by that disk, and will get an optimized VM
+instance for this specific combination.
+
+This document is focused on differentiating the different mechanics and how
+they play together to support the _start small_ (or sparse VM definition)
+use-case.
+
+This topic will be visited from two sides:
+1. Compose - How the user can put together a VM definition
+2. Complete - How the system transforms this potentially partial definition
+   into a useful one
+
+
+## Requirements
+There are certain requirements which should be met 
+
+* Allow creation of building blocks
+* Allow setting defaults for certain expansions/inferences
+* Allow capturing the output of Config post-processing
+
+
+## Compose
+
+### User Config
+
+The user starts with writing a minimal VM configuration.
+
+An example:
+
+```yaml
+kind: Config
+metadata:
+  labels:
+    distro: suse-10
+    instanceType: small
+spec:
+  domain:
+    devices:
+      disk:
+        claimName: my-suse
+```
+
+This definition is completly user driven.
+
+
+### Presets
+
+The first way how this user provided config can be extended is by using
+presets.
+Conceptualy presets work like Kubernetes Pod
+[Presets](https://kubernetes.io/docs/tasks/inject-data-application/podpreset/).
```

Comment:
+1 on clarifying the language and separating terms.

Let me note that some are more user-facing features to me (flavors), and other just mechanism to enable those (presets).

But I'll continue at #305.